src/lib.rs

```diff
@@ -151,40 +152,80 @@ struct HelperInner {
     consumer_done: bool,
 }
 
-/// Error type for `from_env` function.
+/// Return type for `from_env_ext` function.
 #[derive(Debug)]
-pub enum ErrFromEnv {
-    /// There isn't env var, that describes jobserver to inherit.
-    IsNotConfigured,
-    /// Cannot connect following this process's environment.
-    PlatformSpecific {
-        /// Error.
-        err: io::Error,
-        /// Name of gotten env var.
-        env: &'static str,
-        /// Value of gotten env var.
-        var: String,
-    },
+pub struct FromEnv {
+    /// Result of trying to get jobserver client from env.
+    pub client: Result<Client, FromEnvError>,
+    /// Name of the environment variable.
+    /// Empty if no relevant environment variable is found.
+    pub var_name: &'static str,
+    /// Value of the environment variable.
+    /// Empty if no relevant environment variable is found.
+    pub var_value: OsString,
 }
 
-impl std::fmt::Display for ErrFromEnv {
+/// Error type for `from_env_ext` function.
+#[derive(Debug)]
+#[non_exhaustive]
+pub enum FromEnvError {
+    /// There is no environment variable that describes jobserver to inherit.
+    NoEnvVar,
+    /// Cannot parse jobserver environment variable value, incorrect format.
+    CannotParse(String),
+    /// Cannot open path or name from the jobserver environment variable value.
+    CannotOpenPath(String, io::Error),
```

Comment:
```suggestion
    CannotOpenPath(PathBuf, io::Error),
```

Reply:
Sounds reasonable , I thought about using OsString for a moment but then you would still need to convert to string in formatting.

Reply:
Actually maybe it's reasonable to make it `OsString` because the env var values are `OsString`.

It's hard to parse `OsString`s without converting to `String` right now because methods like `contains` or `starts_with` are lacking, but it would be more future compatible.